coderd/users_test.go

```diff
@@ -374,6 +374,103 @@ func TestDeleteUser(t *testing.T) {
 	})
 }
 
+func TestNotifyDeletedUser(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OwnerNotified", func(t *testing.T) {
+		t.Parallel()
+
+		// given
+		notifyEnq := &testutil.FakeNotificationsEnqueuer{}
+		adminClient := coderdtest.New(t, &coderdtest.Options{
+			NotificationsEnqueuer: notifyEnq,
+		})
+		firstUser := coderdtest.CreateFirstUser(t, adminClient)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		user, err := adminClient.CreateUser(ctx, codersdk.CreateUserRequest{
+			OrganizationID: firstUser.OrganizationID,
+			Email:          "another@user.org",
+			Username:       "someone-else",
+			Password:       "SomeSecurePassword!",
+		})
+		require.NoError(t, err)
+
+		// when
+		err = adminClient.DeleteUser(context.Background(), user.ID)
+		require.NoError(t, err)
+
+		// then
+		require.Len(t, notifyEnq.Sent, 2)
+		// notifyEnq.Sent[0] is create account event
+		require.Equal(t, notifications.TemplateUserAccountDeleted, notifyEnq.Sent[1].TemplateID)
```

Comment:
Should the owner receive an email for their own action?
We're suppressing this for users who delete their own workspaces, for example, so I wonder if we should follow the same pattern. Possible follow-up though, non-blocking.

Reply:
You can't delete your own account, so I believe there is no need to filter.

Reply:
What i mean is: if an owner initiated an event - they should probably not receive a notification about it because it's redundant.

Reply:
I understand that they won't receive as the API method will fail faster? see [code](https://github.com/coder/coder/blob/4242fd9c1b1c9ce696a465af5093adf38d1e6b2a/coderd/users.go#L533)